codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parsers/SerializeFunctionNamer.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.rust.codegen.smithy.protocols.parsers
+
+import software.amazon.smithy.codegen.core.SymbolProvider
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.MemberShape
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.Shape
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.model.shapes.UnionShape
+import software.amazon.smithy.rust.codegen.util.toSnakeCase
+
+object SerializeFunctionNamer {
```

Comment:
I would probably also have this be `InlineFunctionNamer` and have it accept a prefix since it's not just for `SerializeFunctions`

Reply:
the same issue will exist for deserializers. Also, it's actually really handy to have the function name prefix easy to grep for in the code so I think doing the `serialize_operation` prefix here might actually make it a little harder to work with